vars/buildDockerAndPublishImage.groovy

```diff
@@ -9,133 +9,119 @@ def call(String imageName, Map config=[:]) {
+            withEnv(["HADOLINT_REPORT=${hadolintReport}"]) {
+              try {
+                sh 'make lint'
+              } finally {
+                recordIssues(
+                  enabledForFailure: true,
+                  aggregatingResults: false,
+                  tool: hadoLint(id: "hadolint-${dockerImageName.replaceAll('/','-')}", pattern: hadolintReport)
+                )
               }
             }
+          } // stage
+
+          stage("Build ${dockerImageName}") {
+            sh 'make build'
+          } //stage
+
+          // There can be 2 kind of tests: per image and per repository
+          [
+            "Image Test Harness": "${dockerImageDir}/cst.yml",
+            "Common Test Harness": "${env.WORKSPACE}/common-cst.yml"
+          ].each { testName, testHarness ->
+            if (fileExists(testHarness)) {
+              stage("Test ${testName} for ${dockerImageName}") {
+                withEnv(["TEST_HARNESS=${testHarness}"]) {
+                  sh 'make test'
+                } // withEnv
+              } //stage
+            } // if
           }
-          echo "Next Release Version = $nextVersion"
-        } // steps
-      } // stage
-
-      stage('Prepare') {
-        environment {
-          DOCKER_REGISTRY = credentials("${dockerConfig.credentials}")
-        }
-        steps {
-          sh 'echo "${DOCKER_REGISTRY_PSW}" | img login -u "${DOCKER_REGISTRY_USR}" --password-stdin'
-          script {
-            // Retrieve Makefile
-            writeFile file: 'Makefile', text: makefileContent
-          }
-        } // steps
-      } // stage
-
-      stage("Lint") {
-        steps {
-          sh 'make lint'
-        } // steps
-        post {
-          always {
-            recordIssues enabledForFailure: true, tool: hadoLint(pattern: 'hadolint.json')
-          }
-        } // post
-      } //stage
-
-      stage("Build") {
-        steps {
-          sh 'make build'
-        } // steps
-      } //stage
-
-      stage("Test") {
-        when {
-          expression { fileExists 'cst.yml' }
-        }
-        steps {
-          sh 'make test'
-        } // steps
-      } //stage
-
-      stage("Semantic Release") {
-        when {
-          allOf {
-            expression { env.SEMANTIC_RELEASE == 'true' }
-            branch dockerConfig.mainBranch  
-          }
-        }
-        steps {
-          echo "Configuring credential.helper"
-          sh 'git config --local credential.helper "!f() { echo username=\\$GIT_USERNAME; echo password=\\$GIT_PASSWORD; }; f"'
 
-          withCredentials([usernamePassword(credentialsId: "${dockerConfig.gitCredentials}", passwordVariable: 'GIT_PASSWORD', usernameVariable: 'GIT_USERNAME')]) {
-            echo "Tagging New Version: $nextVersion"
-            sh "git tag $nextVersion"
+          // Automatic tagging on principal branch is not enabled by default
+          if (semVerEnabled) {
```

Comment:
this will fail if enabled, and more than one image is defined.

Reply:
Good point. I'm not sure on how to prevent this, as I don't see a way to now if the end user is calling other instances of the function. WDYT?

Reply:
I would just update the comment with Gareth input and move forward

Reply:
We don't have that need today and we can still improve the process once that need raise. Commenting the behavior will help in the futur